src/components/DataControllerMixin/DataControllerMixin.js

```diff
@@ -38,12 +38,27 @@ var ModalMixin = {
 
 
   componentWillReceiveProps: function (nextProps) {
-    this.setState({
-      timeSeriesDatasetId: nextProps.meta[0].unique_id,
-    });
-    if (this.verifyParams(nextProps)) {
+    if (this.verifyParams(nextProps) && nextProps.meta.length > 0) {
+      this.setState({
+        timeSeriesDatasetId: nextProps.meta[0].unique_id,
+      });
       this.getData(nextProps);
     }
+    else { //Didn't receive any valid data.
+      //Most likely cause in production would be the user selecting
+      //parameters (rcp, model, variable) for which no datasets have been
+      //added to the database yet.
+      //In development, could be API or ensemble misconfiguration, database down.
+      //Display an error message on each viewer in use by this datacontroller.
+      var text = "No data matching selected parameters available";
+      var viewers = ["StatsTable", "ClimoSeries", "TimeSeries"];
+      _.each(viewers, function(viewer) {
+        var viewerFunction = `set${viewer}NoDataMessage`;
+        if(typeof this[viewerFunction] == 'function'){
+          this[viewerFunction](text);
+        }
```

Comment:
Wow, that is some JS kung fu. Although it's repetitive, and clearly that's the point of your code, I might consider just indexing a hash object mapping viewer name to function directly, thus:

```
viewerFunction = {
   'name1': this.setname1NoDataMessage,
   ...
}[viewer]
```

Reply:
Thanks for your suggestion! I used it as a starting point, but ended up doing something slightly different, trying to minimize repetition. Would appreciate your stylistic feedback if you have a chance. :)